docs/reverse_proxy.md

```diff
@@ -98,6 +98,37 @@ example.com:8448 {
   reverse_proxy http://localhost:8008
 }
 ```
+[Delegation](delegate.md) example:
+```
+(matrix-well-known-header) {
+    # Headers
+    header Access-Control-Allow-Origin "*"
+    header Access-Control-Allow-Methods "GET, POST, PUT, DELETE, OPTIONS"
+    header Access-Control-Allow-Headers "Origin, X-Requested-With, Content-Type, Accept, Authorization"
+    header Content-Type "application/json"
+}
+
+example.com {
+    handle /.well-known/matrix/server {
+        import matrix-well-known-header
+        respond `{"m.server":"matrix.example.com:8448"}`
+    }
+
+    handle /.well-known/matrix/client {
+        import matrix-well-known-header
+        respond `{"m.homeserver":{"base_url":"https://matrix.example.com"},"m.identity_server":{"base_url":"https://identity.example.com}}`
+    }
+}
+
+matrix.example.com {
+    reverse_proxy /_matrix/* http://localhost:8008
+    reverse_proxy /_synapse/client/* http://localhost:8008
+}
+
+matrix.example.com:8448 {
```

Comment:
I'm pretty sure you have to prefix this with `https://`? Can you test this and confirm it works?

Also, making sure that it only sends federation-related paths to this handler, and only client-related paths to the handler above, would help with illustrating the difference between these.

Reply:
> I'm pretty sure you have to prefix this with `https://`? Can you test this and confirm it works?

You don't have to specify `https://` as that's implicit. As per https://caddyserver.com/docs/caddyfile/concepts#addresses which I linked in the description as well (for this very reason actually so my reasoning for choosing this configuration is explained).

Reply:
> Also, making sure that it only sends federation-related paths to this handler, and only client-related paths to the handler above, would help with illustrating the difference between these.

Yes, I agree. I'll add it.